### ISSUE TITLE
[FEATURE REQUEST] It is necessary to add proxy settings

### ISSUE THREAD

**Issue by heheda123123** [marticliment/UniGetUI#1016]
## Before submitting the issue

- [x] **I have checked that the requested feature is not a duplicate or, if it is, it needs to be reopened**
- [x] **I have checked that I am running the latest version of WingetUI**
- [x] **I have checked the [FAQ](https://github.com/marticliment/WingetUI#frequently-asked-questions) section**
- [x] **I have read the [contributing guidelines](https://github.com/marticliment/WingetUI/blob/main/CONTRIBUTING.md#feature-requests) and I agree with the [Code of Conduct](https://github.com/marticliment/WingetUI/blob/main/CODE_OF_CONDUCT.md)**

## Is your feature request related to a problem? Please describe.

In some countries, accessing scoop choco winget is already very slow, and you need to set a proxy when you use it normally. I hope wingetui can add the proxy setting function.
At the same time, wingetui itself also needs to access github, and it is very slow without proxy.

## Describe the solution you'd like

add the proxy setting, Specify proxy settings when running the specified tool in the background commandline.
And use the proxy setting when WingetUI itsself access github.

**Comment by heheda123123** [comment]
When wingetui is started, github will be accessed, I guess it is to detect updates. If you use tools such as proxifier, you can proxy this traffic. But the system proxy is used when using winget to install software in wingetui. It is best to add a proxy setting function, so that the proxy of various tools and wingetui itself can be used uniformly.
Now every time I use wingetui I need to turn on the global proxy in clash and proxifier.

**Comment by marticliment** [comment]
The issue with this feature is how external tools (winget.exe, choco,exe) can be configured to follow the wingetui proxy

**Comment by coolxitech** [comment]
Is there any other way besides manually setting proxy settings for related software?

**Comment by heheda123123** [comment]
> Is there any other way besides manually setting proxy settings for related software?除了手动设置相关软件的代理设置之外，还有其他方法吗？

use proxifier

**Comment by coolxitech** [comment]
> > Is there any other way besides manually setting proxy settings for related software?除了手动设置相关软件的代理设置之外，还有没有其他方法？
> 
> use proxifier使用防腐剂

不如clash

**Comment by heheda123123** [comment]
> > > Is there any other way besides manually setting proxy settings for related software?除了手动设置相关软件的代理设置之外，还有没有其他方法？
> > 
> > 
> > use proxifier使用防腐剂
> 
> 不如clash

clash得开tun才行，没proxifier灵活

**Comment by eizedev** [comment]
It would be great if proxy support for the package providers would be added. f.e. if configured a proxy server in the ui settings then it will also be forwarded to the package provider clis. or that WingetUI and the providers used by wingetui should support system proxy settings. Most of the providers have build in proxy support (http/socks, user/pass ...)

Examples:  
pip: `python.exe -m pip install --upgrade pip --proxy 'http://proxy.example.org:8080'`  
scoop: `scoop config proxy proxy.example.org:8080`  
choco: `choco download somepkg --proxy="'value'" --proxy-user="'<user>'" --proxy-password="'<pwd>'" --proxy-bypass-list="'<comma separated, list>'" --proxy-bypass-on-local`

**Comment by marticliment** [comment]
I will study that, it looks promising

**Comment by CypressVillage** [comment]
since v1.8, winget-cli support proxy, too

`winget settings --enable ProxyCommandLineOptions`

`winget install --id Microsoft.Powershell --source winget --proxy http://127.0.0.1:7890`

**Comment by anjia0532** [comment]
Thanks for @CypressVillage .
1. `winget settings --enable ProxyCommandLineOptions`
2. Custom proxy by cli flags `--proxy `
![Image](https://github.com/user-attachments/assets/f1161248-77f6-4840-9869-ed581ba12210)

**Comment by heheda123123** [comment]
It would be better to uniget to set up the proxy for the various command-line tools in a unified manner

**Comment by treysis** [comment]
@heheda123123 not really, because e.g. winget can just use the system proxy settings. Now if you configure custom proxy settings in UniGetUI it breaks winget because proxy settings are only supported if run as admin (which is neither always possible nor wanted).
